pkg/kubectlcmd/client.go

```diff
@@ -89,14 +89,38 @@ func (c *Client) Apply(dryRun, verbose bool, kubeconfig, context, namespace stri
 	log.Infof("applying to namespace %s:\n%s\n", namespace, cmdStr)
 
 	err := c.cmdSite.Run(cmd)
+	csError := consolidateError(stderr.String())
+
 	if err != nil {
 		logAndPrint("error running kubectl apply: %s", err)
-		return stdout.String(), stderr.String(), fmt.Errorf("error running kubectl apply: %s", err)
+		return stdout.String(), csError, fmt.Errorf("error running kubectl apply: %s", err)
 	}
 
 	logAndPrint("kubectl apply success")
 
-	return stdout.String(), stderr.String(), nil
+	return stdout.String(), csError, nil
+}
+
+// consolidateError is a helper function to dedup the error messages
+func consolidateError(errorMessage string) string {
+	errorMap := make(map[string]int)
+	stderrSlice := strings.Split(errorMessage, "\n")
+	for _, item := range stderrSlice {
+		_, exist := errorMap[item]
+		if exist {
+			errorMap[item] ++
+		} else {
+			errorMap[item] = 1
+		}
+	}
+	var sb strings.Builder
+	for msg, count := range errorMap {
+		if msg == "" {
+			continue
+		}
+		sb.WriteString(fmt.Sprintf("%s (repeated %v times)\n", msg, count))
```

Comment:
I think this will reorder the resulting text block. It's better to preserve the original ordering.

Reply:
Assume that one line of error corresponds to one k8s object. If we want to keep the original ordering in this case, then we need to only suppress consecutive errors, right? However, even though we suppress consecutive errors only, the original ordering becomes less meaningful, because it is hard to match the consolidated from of errors with original k8s manifests

Reply:
Determinism is important in such cases because people expect it. It's confusing to users to run a command twice and see two different outputs for the same input.

Reply:
ok I updated the logic to make the output consistent ordered.